File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.go
type Item struct{
    val int
    idx int
}

func dailyTemperatures(temperatures []int) []int {
    //monotonic stack
    stack := list.New()
    result := make ([]int, len(temperatures))

    for i:=0; i< len(temperatures);i++{
        item := &Item{val: temperatures[i], idx : i}
        for stack.Len() != 0{
            top := stack.Back().Value
            topItem := top.(*Item)

            if !(temperatures[i]> topItem.val){
                break
            }
            top = stack.Remove(stack.Back())
            topItem = top.(*Item)
            result[topItem.idx]=i - topItem.idx
        }
        stack.PushBack(item)
    }
    return result
}

File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> ans(temperatures.size(), 0);
        stack<int> st;
        int maxT = 0;
        for (int i = 0; i < temperatures.size(); i++) {
            cout << temperatures[i] << "\n";
            while ( !st.empty() && temperatures[i] > temperatures[st.top()] ){
                int idx = st.top();
                st.pop();
                ans[idx] = i - idx;
            }
            st.push(i);

            
        }
        return ans;
    }
};